### ISSUE TITLE
[STORM-3292] Trident HiveState must flush writers when the batch commits

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#7074]
<p>For trident the hive writer is flushed only after it hits the batch size.</p>

<p>see - <a href="https://github.com/apache/storm/blob/master/external/storm-hive/src/main/java/org/apache/storm/hive/trident/HiveState.java#L108" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/blob/master/external/storm-hive/src/main/java/org/apache/storm/hive/trident/HiveState.java#L108</a></p>

<p>Trident HiveState does not flush during the batch commit and it appears to be an oversight. Without this trident state cannot guarantee at-least once. (E.g. if the transaction is open but trident moves to the next txid and later fails the data in the open transaction is lost).</p>

<p>So I think for at-least once, the HiveState must flush all the writers irrespective of the batch sizes when trident invokes the "commit(txid)" .</p>

---
<details><summary><i>Originally reported by <a title="Arun Mahadevan" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=arunmahadevan">arunmahadevan</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-3292" target="_blank">Trident HiveState must flush writers when the batch commits</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Arun Mahadevan" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=arunmahadevan">arunmahadevan</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-12-02T07:18:03+00:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>:</i>
<p>Thanks <a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=arunmahadevan" class="user-hover" rel="arunmahadevan">Arun Mahadevan</a>, I merged into master and 1.x-branch.</p>
